### ISSUE TITLE
multiplier slow converge

### ISSUE THREAD

**Issue by hailinzeng** [dawsonjon/fpu#14]
Hi, Jon

I just found another interesting trace that multiplier is stuck at state normalize_2 for a long time.

input:
a : 32'h5e_c72e (denormal)
b : 32'h1f_ddeb (denormal)

![waveform](https://raw.githubusercontent.com/exa-mining/oss-c-rtl/master/ieee754/denormal_slow/waveform.png)

I have uploaded the testbench in [here](https://github.com/exa-mining/oss-c-rtl/blob/c9816d5042ce8996713579b614eb2e11956abb6f/ieee754/denormal_slow/testbench.v#L1-L498).

**Comment by hailinzeng** [comment]
In this trace, z_e is 0x302 when state becomes `normalize_2`. $signed(z_e) is -254, which requires 128 cycles to change the state to `round`. 

https://github.com/dawsonjon/fpu/blob/97a768293e9376aed6ee9003f41d68c7de71cf53/multiplier/multiplier.v#L209-L220

It may not be considered as a bug. But it seems to be a good target for optimization.

**Comment by dawsonjon** [comment]
Yes, I think that could be a good target for optimisation. So I will leave this issue open for now as a reminder. However if you are looking for some *fast* code... you could take a look at [my other project](verilog-math/components/mul.v) which provides a fully pipelined multiplier with a 9 clock cycle latency.
